### ISSUE TITLE
6.12.0 package includes tests

### ISSUE THREAD

**Issue by Tecuya** [jazzband/pip-tools#1778]
The 6.12.0 release is installing a `tests` package containing the tests for pip-tools.

This means any project that has pip-tools installed will have its own `tests` modules eclipsed by pip-tools' `tests`.  This broke our CI build when I upgraded all the deps in a project today, chased it down to pip-tools.  

Demonstration:
```
root@docker-fi:~# pip install pip-tools==6.11.0
[...snip...]
Successfully installed pip-tools-6.11.0
root@docker-fi:~# ls /usr/local/lib/venv/lib/python3.9/site-packages/tests
ls: cannot access '/usr/local/lib/venv/lib/python3.9/site-packages/tests': No such file or directory
root@docker-fi:~# pip install pip-tools==6.12.0
[...snip...]
Successfully installed pip-tools-6.12.0
root@docker-fi:~# ls /usr/local/lib/venv/lib/python3.9/site-packages/tests
__init__.py  conftest.py   test_cache.py	test_cli_sync.py  test_fake_index.py  test_minimal_upgrade.py	test_repository_pypi.py  test_subprocess_utils.py  test_top_level_editable.py  test_writer.py
__pycache__  constants.py  test_cli_compile.py	test_data	  test_logging.py     test_repository_local.py	test_resolver.py	 test_sync.py		   test_utils.py	       utils.py
root@docker-fi:~# 
```

**Comment by webknjaz** [comment]
Just to clarify the expectations:
1) the sdist must bundle tests
2) the wheel should avoid including them

P.S. I haven't checked but it might be related to the updates in setuptools. Whoever ends up debugging this could try to verify this by running `PIP_CONSTRAINT=build-constraints.txt python -m build` with various setuptools versions pinned in `build-constraints.txt` and using `wheel unpack` to inspect the contents.
Checking the build log output can also shed some light on the problem.

**Comment by tnlogy** [comment]
I also have some issues with this, now I cannot import my local package `from tests.utils import ...` since pip-tools tests directory have a `utils.py` file in it.

**Comment by AndydeCleyre** [comment]
The current wheel is also polluting `site-packages` with `docs` and `examples`.

**Comment by q0w** [comment]
But the default autodiscovery provided by setuptools does not work for pip-tools (neither src layout nor flat layout), so maybe provide explicity packages for setuptools.packages.find or exclude explicitly tests and other packages

**Comment by atugushev** [comment]
The fix is released as part of `pip-tools v6.12.1`.
